[Infrastructure UI]: Implement Telemetry events for Hosts View

tonyghiani opened the pull request:
## 📓 Summary

Closes #148739 

Implement the following Telemetry custom events:
- **Host Entry Clicked**
- **Hosts View Query Submitted**

When triggered, these events will track to FullStory some details relevant to the performed query or the selected host from the table.

These changes include a necessary refactor of the hosts' table hook in order to trigger an event on each entry click without performance issues.

## 🧪 Testing

As long as we don't have access to the staging environment of FullStory, is difficult to test E2E if an event is tracked or not.

What we can firstly test is that our events are correctly collected by the `core.analytics` API and shipped correctly to the FullStory integration. To verify this, please follow these steps:

1. Add the following configuration to your `kibana.dev.yml` file:
```yml
# Enable Fullstory Telemetry
xpack.cloud.id: "elastic_kibana_dev"
xpack.cloud.full_story.enabled: true
# Staging FullStory OrgID
xpack.cloud.full_story.org_id: "1397FY"
xpack.cloud.full_story.eventTypesAllowlist: ["Hosts View Query Submitted", "Host Entry Clicked"]
```

2. Add debugging breakpoint into the [`sendToShipper()` function](https://github.com/elastic/kibana/blob/main/packages/analytics/client/src/analytics_client/analytics_client.ts#L275) in the analytics package.
3. Navigate to Hosts View and perform search queries or click on table entries to check whether the events are correctly composed.

Commit:
refactor(infra): useHostsTable and event tracking

Commit:
feat(infra): track event on query submission

Commit:
feat(infra): track event on query submission

elasticmachine commented:
Pinging @elastic/infra-monitoring-ui (Team:Infra Monitoring UI)

tonyghiani committed:
Merge branch 'main' into 148739-hosts-telemetry-tracking

juliaElastic committed:
[Fleet] added back batch exec for update tags (#148618)

## Summary

Closes https://github.com/elastic/kibana/issues/148233

Fixing issue of bulk update tags not working with the new agent status
runtime field.
Refactored update tags to use batching again, resolving 10k agent ids at
a time and `updateByQuery` on batches.
Works locally with dummy agents, has to be tested on cloud with horde to
simulate conflicts.


### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

shahzad31 committed:
[Synthetics] Object types panel and thresholds (#149099)

Fixes https://github.com/elastic/kibana/issues/145395
Fixes https://github.com/elastic/kibana/issues/145892

dej611 committed:
[Lens] Enable previous time shift when using a date histogram (#149126)

## Summary

Fixes #104259 

This PR expands the support of the `previous` time shift for
visualisations with a date histogram defined.
On the implementation side, an absolute time shift in the shape of
`endAt( startRange )` is leveraged to make it work the `previous` shift:
anchoring the shift to the beginning of the current range will make sure
to compute the right shift in terms of buckets in order to avoid
misaligned shifts and the main reason why this feature has been disabled
initially.

I've tried to condense an explanation here with this diagram of the
misalignment problem:


![previous_before](https://user-images.githubusercontent.com/924948/213234848-40e3382a-843d-43fa-83d1-769a5f2e7953.png)

With the current approach, there's a small time range overlap between
the two requested interval, but that's the result of the rounding logic
to get both shifts aligned.


![previous_2](https://user-images.githubusercontent.com/924948/213235429-1c99aefb-e18b-450a-b2d7-45b7d74c9e71.png)

The only alternative to avoid the overlap is to get a gap between the
two, but the former seems a better result to me.

### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] Any UI touched in this PR is usable by keyboard only (learn more
about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures
(run axe in browser:
[FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/),
[Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive
layout. (You can test this [in your
browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser
compatibility](https://www.elastic.co/support/matrix#matrix_browsers)


### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to
identify risks that should be tested prior to the change/feature
release.

When forming the risk matrix, consider some of the following examples
and how they may potentially impact the change:

| Risk | Probability | Severity | Mitigation/Notes |

|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space.
| Low | High | Integration tests will verify that all features are still
supported in non-default Kibana Space and when user switches between
spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions
when multiple Kibana nodes are polling for the same tasks. | High | Low
| Tasks are idempotent, so executing them multiple times will not result
in logical error, but will degrade performance. To test for this case we
add plenty of unit tests around this logic and document manual testing
procedure. |
| Code should gracefully handle cases when feature X or plugin Y are
disabled. | Medium | High | Unit tests will verify that any feature flag
or plugin combination still results in our service operational. |
| [See more potential risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers

- [ ] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

Co-authored-by: Stratoula Kalafateli <efstratia.kalafateli@elastic.co>

JordanSh committed:
[Cloud Posture] CIS AWS support - changes to findings tables (#148945)

awahab07 committed:
[Synthetics] Better formatting for waterfall timeline tooltips (#149142)

Co-authored-by: Shahzad <shahzad31comp@gmail.com>

walterra committed:
[ML] Consolidate `query_utils` into package `@kbn/ml-query-utils` (#149224)

Consolidates more `query_utils` into package `@kbn/ml-query-utils`.

droberts195 committed:
[ML] Remove job_type from job definitions in modules (#149247)

All the job definitions in ML modules contain the field `job_type`. For
anomaly detection jobs this is always `anomaly_detector`. Officially it
is not a valid input to the create job API. Elasticsearch accepts it
(which is technically a bug as it's out-of-spec) which is why it has
never been noticed before. However, the language clients do not accept
it, which causes problems if somebody wants to create module jobs using
a language client. It's best that the jobs in our modules conform
accurately to the specification of a valid job creation request.

hop-dev committed:
[Fleet] Use optimistic locking when updating `installed_es` on input package policy creation (#148883)

## Summary

Part of #145529 

When creating an input package policy, we now create index templates and
ingest pipelines. As part of this operation we have to update
installed_es on the installation saved object, there is a risk of lost
updates if multiple package policies are created at the same time, to
combat this i have used the in built saved object optimistic
concurrency.

I have tested this locally, we do start see conflicts occur if I create
500 package policies in concurrent batches of 25, but I think we should
have a dedicated bulk endpoint if we want to handle more than that.

I haven't pushed the automated tests as they take a few minutes to run
and I don't think there is a big enough benefit to running them as part
of CI every time.

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

rudolf committed:
Bump elasticsearch-js to 8.6.0-canary.3 (#148521)

## Summary

Bump elasticsearch-js to 8.6.0-canary.3 to unblock
https://github.com/elastic/kibana/issues/145653

The updated version of elasticsearch-js comes with some type changes
that causes typescript type checking to fail. I've fixed the type errors
that were obvious/easy but left todo's for some types which were harder
for me to figure out. If any of these todo's are in your team's code,
please contribute directly to the branch to fix them.


### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] Any UI touched in this PR is usable by keyboard only (learn more
about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures
(run axe in browser:
[FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/),
[Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive
layout. (You can test this [in your
browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser
compatibility](https://www.elastic.co/support/matrix#matrix_browsers)


### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to
identify risks that should be tested prior to the change/feature
release.

When forming the risk matrix, consider some of the following examples
and how they may potentially impact the change:

| Risk | Probability | Severity | Mitigation/Notes |

|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space.
| Low | High | Integration tests will verify that all features are still
supported in non-default Kibana Space and when user switches between
spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions
when multiple Kibana nodes are polling for the same tasks. | High | Low
| Tasks are idempotent, so executing them multiple times will not result
in logical error, but will degrade performance. To test for this case we
add plenty of unit tests around this logic and document manual testing
procedure. |
| Code should gracefully handle cases when feature X or plugin Y are
disabled. | Medium | High | Unit tests will verify that any feature flag
or plugin combination still results in our service operational. |
| [See more potential risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers

- [ ] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Thom Heymann <thom.heymann@elastic.co>
Co-authored-by: weltenwort <stuermer@weltenwort.de>

ofiriro3 committed:
[Cloud Posture] - Deprecate csp rule remove migration object (#148530)

afharo committed:
[FTRs] Execution Context: fix no data flakiness (#149406)

Resolves https://github.com/elastic/kibana/issues/112102

crespocarlos commented on x-pack/plugins/infra/public/pages/metrics/hosts/components/metrics_trend/metrics_chart.tsx:
```diff
@@ -120,6 +120,6 @@ export const MetricsChart = ({
 const ChartStyled = styled(Chart)`
   .echMetric {
-    border-radius: 5px;
+    border-radius: ${(p) => p.theme.eui.euiBorderRadius};
```
sweet!

crespocarlos commented on x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_hosts_table.tsx:
```diff
@@ -0,0 +1,196 @@
+/**
+ * Columns translations
+ */
+const titleLabel = i18n.translate('xpack.infra.hostsTable.nameColumnHeader', {
```
Out of curiosity: why did you separate the translations from the columns array?

tonyghiani replied:
Being those translations part of the `columns` list, every time the `time` updates it implies recomputing those same static translations invoking the `i18n.translate()` function for each column.

Moving those static texts out of the functions, they are computed once at module time and we'll save various steps in the call stack on each column recomputing.

I myself consider this a micro-optimization, but it's worth leaving it clean and split :) Let me know your thoughts and if you see it better in the function.

crespocarlos replied:
Yeah, it makes sense. I like it. I was thinking more about the trade-off between readability and actual gain from the optimization, because the total nr of lines roughly doubled after it. But the file is still small, so I wouldn't worry about it now.

crespocarlos commented on x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_unified_search.ts:
```diff
@@ -6,21 +6,34 @@
+  return {
+    control_filters: panelFilters.map((filter) => JSON.stringify(filter)),
+    filters: filters.map((filter) => JSON.stringify(filter)),
+    interval: `interval(from:${dateRange.from},to:${dateRange.to})`,
```
I wonder if we can to send the actual interval, like 15m, 1h, 20days instead of only `from` and `to`

```
moment.duration(moment(dateRange.to).diff(moment(dateRange.from))).humanize()
```

shahzad31 committed:
[Synthetics] Errors list active state (#149387)

Fixes https://github.com/elastic/kibana/issues/145373

crespocarlos commented on x-pack/plugins/infra/public/pages/metrics/hosts/hooks/use_unified_search.ts:
```diff
@@ -6,21 +6,34 @@
+  const { panelFilters, filters, dateRange, query: queryObj } = hostState;
+
+  return {
+    control_filters: panelFilters.map((filter) => JSON.stringify(filter)),
+    filters: filters.map((filter) => JSON.stringify(filter)),
```
`Filter` type has `state$` and `meta` properties that we might not be interested in, for telemetry purposes. 

How about sending only the query? 

```
control_filters: JSON.stringify(panelFilters.map((filter) => (filter.query)),
filters: JSON.stringify(filters.map((filter) => filter.query))),
```

This way, we'll get something like:

```
[{
  match_phrase: { host.name: "metricbeat" }
}]
````

tonyghiani replied:
This was my first attempt, I had the same opinion as you! However, I noticed in some cases there are details relevant to the Filter which are not in the query.
For example, I noticed that a negated control panel is not marked in the query, but adds a `meta` field `negated: true`, that's why I saw the need to include the whole filter to avoid misleading filters.

crespocarlos replied:
I see. Indeed.I don't know how useful that information actually is, but let's try sending the whole object and see how it goes.

Commit:
[APM] Allow calling `createInternalESClient` without `context` (#149320)

crespocarlos reviewed (commented):
This is definitely a step towards the right path in regards to telemtry. Good job making it easier to trigger custom events.

I left a couple of comments and we might have to add these custom events to Fullstory allow list.

sabarasaba committed:
[Index Management] fix flaky test (#149384)

kpollich committed:
[Fleet] Fix parsing logic for data_stream.hidden manifest field (#149357)

## Summary

Fixes package archive parsing logic for the
`index_template_data_stream.hidden` property, implemented in line with
other `index_template.*` "dotted property name" values from package
manifests.

## To test

Create a test package that includes
`elasticsearch.index_template.data_stream.hidden: true` in a
`data_stream/**/manifest.yml` file, install the package, then verify
(via dev tools - NOT stack management!!!) that the generated index
template contains `data_stream.hidden: true`


![image](https://user-images.githubusercontent.com/6766512/214155715-1a78a7f6-cde0-4cb1-a6e3-fd1ac3749f1e.png)

It'd be great to have automated tests around this, but testing our
archive parsing logic is very challenging due to the nature of reading
directories/files and how much mocking is required. See
https://github.com/elastic/kibana/issues/147050. I'll see about adding
an integration test for this.

azasypkin committed:
Remove `uglify-js` package.json resolutions. (#149395)

szwarckonrad committed:
[Defend Workflows] [Fix] Test for --help validation  (#149393)

Zacqary committed:
[RAM] Add error for action interval shorter than check interval (#148919)

## Summary

Closes #148569

Adds form error and API error for action intervals shorter than rule
check intervals

<img width="585" alt="Screen Shot 2023-01-13 at 1 46 55 PM"
src="https://user-images.githubusercontent.com/1445834/212406181-059b53e4-a52d-44db-ba03-1a7285c676ce.png">
<img width="592" alt="Screen Shot 2023-01-13 at 1 46 42 PM"
src="https://user-images.githubusercontent.com/1445834/212406188-9b59fc05-45c0-4f93-b68c-c1540d46cd2b.png">


### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

logeekal committed:
[feat][Kibana Presentation] Options List new feature (#149121)

## Background
Security solution recently started using controls plugin to provide
users some extra filtering capability with help of options list control.

During this implementation, there was some feedback was given from
design team + we were facing some minor issues because of the caches.
Below section gives the list of changes and the reasoning behind each
change. All of these changes were discussed with @ThomThomson

## Summary

This PR introduces 3 new functionalities for optionsList embeddables.

1. Cache invalidation option when reloading an optionsList
- In security solution we have transactional alerting system, where user
frequently update alerts data
- We need the latest data and 1 minute cache of optionsList was
preventing us from doing so.
- This change adds the capability to clear the cache from an embeddable.
3. option to hideSearch Panel
- As a client of control plugin, we look some control over what panels
are visible and what are not.
5. Option to add custom placeholder for optionsList
- Out product team felt that default placeholder for optionsList `Any`
may be confusing for the user and hence I have added the option for
clients to provide a placeholder.

### Checklist

Delete any items that are not applicable to this PR.

- [x]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

angorayc committed:
[SecuritySolution] Replace risk score over time with Lens Embeddable (#149035)

## Summary
Implements: https://github.com/elastic/kibana/issues/149015

Please Enable feature flags. Please add this to kibana.dev.yml
xpack.securitySolution.enableExperimental: ['chartEmbeddablesEnabled']

---
### Replace risk score over time with Lens Embeddable

Before:
<img width="2543" alt="Screenshot 2023-01-17 at 10 28 23"
src="https://user-images.githubusercontent.com/6295984/212875145-f39fef08-c152-4c7e-8d0f-cf8e259c0b05.png">

After:
<img width="926" alt="Screenshot 2023-01-24 at 15 03 40"
src="https://user-images.githubusercontent.com/6295984/214329885-71e8166b-07ec-4f09-bece-919189d655ea.png">


### Alerts By severity on host / network / user details should apply
global filters
<img width="1673" alt="Screenshot 2023-01-19 at 11 06 21"
src="https://user-images.githubusercontent.com/6295984/213426977-4b803513-69f4-4074-b45d-2002c3f8fecf.png">

### Styling for donuts on Entity Analytics dashboard
(Moving the legend to left side of the chart so its actions button
wouldn't overlap with chart action)
<img width="1654" alt="Screenshot 2023-01-19 at 11 08 47"
src="https://user-images.githubusercontent.com/6295984/213427320-0fa3a9aa-f0d4-435a-87d3-5108b5c7f991.png">


Preview:
[Host risk score over
time](https://kibana-pr-148624.kb.us-west2.gcp.elastic-cloud.com:9243/s/data/app/security/hosts/name/Angelas-MacBook-Pro.local/hostRisk?sourcerer=(default:(id:security-solution-data,selectedPatterns:!(%27filebeat-*%27,%27logs-*%27,%27packetbeat-*%27,%27-*elastic-cloud-logs-*%27)))&timerange=(global:(linkTo:!(timeline),timerange:(from:%272023-01-18T00:00:00.000Z%27,fromStr:now%2Fd,kind:relative,to:%272023-01-18T23:59:59.999Z%27,toStr:now%2Fd)),timeline:(linkTo:!(global),timerange:(from:%272023-01-18T00:00:00.000Z%27,fromStr:now%2Fd,kind:relative,to:%272023-01-18T23:59:59.999Z%27,toStr:now%2Fd)))&timeline=(activeTab:query,graphEventId:%27%27,id:%2736333270-9731-11ed-a0f5-f16ed1963ee6%27,isOpen:!f))
[User risk score over
time](https://kibana-pr-148624.kb.us-west2.gcp.elastic-cloud.com:9243/s/data/app/security/users/name/angelachuang/userRisk?sourcerer=(default:(id:security-solution-data,selectedPatterns:!(%27filebeat-*%27,%27logs-*%27,%27packetbeat-*%27,%27-*elastic-cloud-logs-*%27)))&timerange=(global:(linkTo:!(timeline),timerange:(from:%272023-01-18T00:00:00.000Z%27,fromStr:now%2Fd,kind:relative,to:%272023-01-18T23:59:59.999Z%27,toStr:now%2Fd)),timeline:(linkTo:!(global),timerange:(from:%272023-01-18T00:00:00.000Z%27,fromStr:now%2Fd,kind:relative,to:%272023-01-18T23:59:59.999Z%27,toStr:now%2Fd)))&timeline=(activeTab:query,graphEventId:%27%27,id:%2736333270-9731-11ed-a0f5-f16ed1963ee6%27,isOpen:!f))

https://p.elstc.co/paste/2MIN+pHd#TETZwPh15r64HQ2z0Cn26Z321XCxe+2DqliqF5-CHmr
Designers' review:
https://github.com/elastic/kibana/issues/149123

### Checklist

Delete any items that are not applicable to this PR.

- [x] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

kc13greiner committed:
Upgrade jsonwebtoken from 8 to 9 (#148128)

## Summary

Upgrading the `jsonwebtoken` library used in
`cloud_integration/cloud_chat` plugin from v8.5 to v9.0

## Migration
After reviewing the affected code it looks like this upgrade shouldn't
cause any breaking changes. Please see the v8 --> v9 migration guide
[here](https://github.com/auth0/node-jsonwebtoken/wiki/Migration-Notes:-v8-to-v9)

## Questions for reviewers
- What is the size and type of `secret` that we are passing into
`generateSignedJwt`?

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Clint Andrew Hall <clint.hall@elastic.co>

kpollich committed:
[Fleet] Flesh out Fleet architecture diagram a bit more (#149420)

## Summary

Adding some more detail to Fleet's dev docs architecture diagram 🙂

mistic committed:
skip flaky suite (#88639)

jbudz committed:
[artifacts/container-image] Trigger image tag update (#149437)

Reopens #149143 with updates to the target file and service

After a commit is merged, tested, and images are built and pushed to the
container registry we need to send a notification that a new tag is
available.

This triggers a promotion pipeline with the latest container tag when:
1) the branch is tracked (i.e. main, and not a personal branch) 1)
~triggered from our on-merge test pipeline.~
https://github.com/elastic/kibana/pull/149350 had to remove support for
this - we're triggering via REST now which removes the from trigger
environment variable.

qn895 committed:
[ML] Fix ML page crash due to css string (#149428)

## Summary

Fix a regression introduced by merge of
https://github.com/elastic/kibana/pull/147322 due to css string not
being wrapped in emotion `css` tag.


![image](https://user-images.githubusercontent.com/43350163/214348591-9f931d0d-8e15-42e4-9a30-65c159e4d149.png)

### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing
guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses
sentence case text and includes [i18n
support](https://github.com/elastic/kibana/blob/main/packages/kbn-i18n/README.md)
- [ ]
[Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html)
was added for features that require explanation or tutorials
- [ ] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [ ] Any UI touched in this PR is usable by keyboard only (learn more
about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures
(run axe in browser:
[FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/),
[Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be
allowlisted in the cloud and added to the [docker
list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive
layout. (You can test this [in your
browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser
compatibility](https://www.elastic.co/support/matrix#matrix_browsers)


### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to
identify risks that should be tested prior to the change/feature
release.

When forming the risk matrix, consider some of the following examples
and how they may potentially impact the change:

| Risk | Probability | Severity | Mitigation/Notes |

|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space.
| Low | High | Integration tests will verify that all features are still
supported in non-default Kibana Space and when user switches between
spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions
when multiple Kibana nodes are polling for the same tasks. | High | Low
| Tasks are idempotent, so executing them multiple times will not result
in logical error, but will degrade performance. To test for this case we
add plenty of unit tests around this logic and document manual testing
procedure. |
| Code should gracefully handle cases when feature X or plugin Y are
disabled. | Medium | High | Unit tests will verify that any feature flag
or plugin combination still results in our service operational. |
| [See more potential risk
examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx) |


### For maintainers

- [ ] This was checked for breaking API changes and was [labeled
appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

jsanz committed:
[Maps] Update EMS resources (#149331)

* Updates to the latest released version of `@elastic/ems-client`
(8.4.0)
* Updates link to EMS landing page to 8.6
* Passes Kibana version to EMS Client instance to request explicitly to
point to the correct API endpoints. This change is introduced to avoid
needing to keep releasing EMS Client with the rest of the stack.

shahzad31 committed:
[Synthetics] Failed tests by step panel (#149322)

## Summary

Failed tests by step panel

Fixes https://github.com/elastic/kibana/issues/145368

<img width="1523" alt="image"
src="https://user-images.githubusercontent.com/3505601/214053308-458c023e-aaab-4db6-8aa4-a18ce150ce19.png">

machadoum committed:
[145663] Refactor explore pages to migrate HoverActions to CellActions (#148056)

Epic: https://github.com/elastic/kibana/issues/144943


## Summary
Update explore pages to use the new `CellActions` component instead of
`HoverActions `.

### What is included?
* Update the user, host, and network page tables.

<img width="1512" alt="Screenshot 2023-01-17 at 13 12 16"
src="https://user-images.githubusercontent.com/1490444/212896520-f41e9026-cef0-4a37-8bd1-35784a87ca09.png">
<img width="1482" alt="Screenshot 2023-01-17 at 13 19 34"
src="https://user-images.githubusercontent.com/1490444/212897411-cd3c3ef8-bca0-461b-a1ff-c7dd67159d1b.png">

* Fields rendered when clicking on "+{N} more" 
<img width="248" alt="Screenshot 2023-01-17 at 12 51 38"
src="https://user-images.githubusercontent.com/1490444/212892255-2ecd7050-75f6-4883-b331-1fed527de53f.png">


### What is NOT included?
* Visualizations
* Fields on details pages. They are also used by the Timeline and need
to be draggable.
* Timeline
* Datagrid tables - Events and Alerts
* Plugins



### Checklist

Delete any items that are not applicable to this PR.

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios
- [x] This renders correctly on smaller devices using a responsive
layout. (You can test this [in your
browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [x] This was checked for [cross-browser
compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

walterra committed:
[ML] Transforms: Use concise fragment syntax. (#149434)

Refactor to use concise fragment syntax (`<>...</>` instead of `<Fragment>...</Fragment>`) where applicable.

paul-tavares committed:
[Security Solution][Endpoint] Adjust the Response Actions file download API so that it is driven by a `file_id` (#149267)

## Summary

- Adjust the file download APIs (info and download) to use a file ID as
a route param instead of having the API calculate the File ID
- Adjusts the UI code to ensure the new APIs are called correctly

shahzad31 committed:
[Synthetics] Test run steps list metrics- content size (#149339)

stephmilovic committed:
[Security solution] Fix IP/geo location in Top N Flow Table (#149369)

dominiqueclarke committed:
[Synthetics] Omit or include `ssl` keys when appropriate for project monitors and private locations (#149298)

## Summary

Resolves https://github.com/elastic/kibana/issues/149083

1. [Prevents tls fields from being
saved](https://github.com/elastic/kibana/pull/149298/files#diff-56296f634bf379eb71629f426c670cd030d2a15263a59964847c0d10af09a767R14)
on the Synthetics Integration policy when `is_tls_enabled` is false
2. Ensures `is_tls_enabled` is set properly for project monitors
([http](https://github.com/elastic/kibana/pull/149298/files#diff-0f42bb3b11a6ab864dee3488d5e9f7282adc009a261b3caee743a880b825c766R73)
and
[tcp](https://github.com/elastic/kibana/pull/149298/files#diff-3ad87e629abc6f17c395e8435c94f0f1a6274c9efea7d24ab81b7635ef0e43dfR69)).
This ensures that when a monitor is sent to a public location or a
private location, the `ssl` fields are sent or stripped appropriately.

### Testing

1. Create a private location
2. Create 2 lightweight project monitors using the following
configuration
```
- type: tcp
  id: 'tls-enabled'
  name: 'TLS-Enabled'
  hosts: ["8.8.8.8:80"]
  ssl:
     verification_mode: 'strict'
```
```
- type: tcp
  id: 'tls-disabled'
  name: 'TLS-Disabled'
  hosts: ["8.8.8.8:80"]
```
3. Set these monitors to execute from both a private and public location
via the `monitor` key in your `synthetics.config.ts` file.
```
    monitor: {
      schedule: 3,
      privateLocations: ["YOUR PRIVATE LOCATION"],
      locations: ["us_central"], // to test against dev environment
    },
```
4. Navigate to the agent policy for the private location and inspect the
full policy. Ensure the Synthetics policy on the agent package policy
does not have `ssl` fields set for ssl disabled monitor. Ensure the
`ssl` fields are set for the ssl enabled monitor.

lcawl committed:
[DOCS] Fix incorrect alt text for images (#149371)

drewdaemon committed:
[Lens] only reload embeddable when a new message was actually added (#149424)

jbudz committed:
[artifacts/container-image] Fix image tag when triggering update (#149449)

Instead of passing the entire qualified tag when triggering an update,
this sets the image tag to the abbreviated commit.

bmorelli25 committed:
[APM] docs: SDH documentation updates (#149359)

## Summary

Two small documentation updates based on feedback in recent SDHs.

Zacqary committed:
[RAM] Fix case conversion of frequency.notify_when in API (#148838)

## Summary

Closes #148572

Fixes an issue where rules APIs required `frequency.notifyWhen` to be
sent/received instead of `frequency.notify_when`


### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

Co-authored-by: Julia <iuliia.guskova@elastic.co>
Co-authored-by: Kibana Machine <42973632+kibanamachine@users.noreply.github.com>

jbudz committed:
skip flaky suite (#103043)

nchaulet committed:
[Fleet] Update openAPI to include agent metrics (#149455)

Commit:
fix triggerActionsUi overage

kibanamachine committed:
[api-docs] 2023-01-25 Daily api_docs build (#149475)

Generated by
https://buildkite.com/elastic/kibana-api-docs-daily/builds/228

dominiqueclarke committed:
[Synthetics]  add test coverage for editing monitors within a non default space (#149138)

dmlemeshko committed:
[api capacity testing] Adjust endpoint limits (#149333)

## Summary

This PR tweaks max load and thresholds for some apis in order to get
more consistent
[results](https://telemetry-v2-staging.elastic.dev/s/kibana-performance/app/dashboards#/view/c4b07f90-58eb-563e-8e59-3be09c8074f8?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-3d,to:now))).

Current main (last 3 days):
<img width="1534" alt="Screenshot 2023-01-24 at 14 53 15"
src="https://user-images.githubusercontent.com/10977896/214314206-0b849053-9473-4a98-9f3e-e90461a5a341.png">

PR (4 runs):

<img width="1569" alt="Screenshot 2023-01-24 at 15 00 49"
src="https://user-images.githubusercontent.com/10977896/214314530-60a5b3e4-2730-4307-946d-cf25d57d04ab.png">

mibragimov committed:
[Console] Use elasticsearch-js's `maxResponseSize` option instead of raw http requests for autocomplete_entities (#148146)

Closes https://github.com/elastic/kibana/issues/145653


### Summary

This PR replaces the raw http requests with the elasticsearch client to
retrieve autocomplete suggestions. In 8.5, and 8.6, the autocomplete
suggestions were retrieved using raw http requests. This was done
because the elasticsearch client's `maxResponseSize` and
`maxCompressedResponseSize` settings were not working as expected when
`Content-Length` header was not set in the response. This was fixed in
https://github.com/elastic/elastic-transport-js/pull/57, so we can now
use the elasticsearch client instead of raw http requests.

Co-authored-by: Muhammad Ibragimov <muhammad.ibragimov@elastic.co>
Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
Co-authored-by: Ignacio Rivas <ignacio.rivas@elastic.co>

crespocarlos committed:
[Infrastructure UI] Add lens charts to hosts view (#148906)

## Summary

closes https://github.com/elastic/obs-infraobs-team/issues/921

This PR introduces a series of lens charts to hosts view page.

![out](https://user-images.githubusercontent.com/2767137/214100703-1847586a-fa87-4c2c-9eb0-af5271667702.gif)

The Lens embeddable component is fully integrated, allowing the filters
to react to clicks on the charts as well as accessing Lens directly from
the chart

The charts will show metrics for:
- Load
- CPU Usage
- RX
- TX
- Memory 
- Disk Read
- Disk Write


### How to test
- Using metricbeat
  - Enable system metric in metricbeat
  - Start your local ES

- Using oblt
  - configure kibana.dev.yml with your oblt-cluster

- Using slingshot
- Clone https://github.com/elastic/slingshot and run slingshot yarn
slingshot load --config ./configs/hosts.json

- Start your local ES
- Start kibana
- Navigate to Infrastructure > Hosts

### Notes
- I've renamed a few things that started to become confusing after this
change.
- The structure created for the lens vis is similar to the
inventory_model, but here is only used in `public` and I've placed them
as a common functionality, since this might also be used in other pages
in the future.

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>

shahzad31 committed:
[Synthetics] Step details page error callout (#149410)

Fixes https://github.com/elastic/kibana/issues/145386

Step details page error callout !!

<img width="1779" alt="image"
src="https://user-images.githubusercontent.com/3505601/214309774-302ca366-31dc-47ee-a863-7452e7d8c848.png">

azasypkin committed:
Remove `node-fetch` package.json resolutions for `puppeteer` package. (#149398)

Commit:
Merge branch '148739-hosts-telemetry-tracking' of github.com:tonyghiani/kibana into 148739-hosts-telemetry-tracking
